SmallCommitsWorkshopTests/Controllers/UsersControllerTests.cs

```diff
@@ -36,28 +42,100 @@ internal sealed class UsersControllerTests {
 							{ "id", user.Id },
 							{ "userName", user.UserName },
+							{ "isActive", user.IsActive },
 						}
 					),
 					await response.Content.ReadAsJsonAsync<IDictionary<long, IDictionary<string, object>>>()
 				);
 			}
 		}
+
+		[Test]
+		public async Task Get_ReturnsUser() {
+			User user = m_defaultUsers.First();
+			using( HttpResponseMessage response = await m_client.GetAsync( $"/api/users/{user.Id}" ) ) {
+				Assert.AreEqual( response.StatusCode, HttpStatusCode.OK );
+				CollectionAssert.AreEquivalent(
+					new Dictionary<long, IDictionary<string, object>>( 2 ) {
+						{
+							user.Id,
+							new Dictionary<string, object>( 2 ) {
+								{ "id", user.Id },
+								{ "userName", user.UserName },
+								{ "isActive", user.IsActive },
+							}
+						}
+					},
+					await response.Content.ReadAsJsonAsync<IDictionary<long, IDictionary<string, object>>>()
+				);
+			}
+		}
+
+		[Test]
+		public async Task CreateOrUpdate_UserDoesNotExist_CreatesUser() {
+			User user = new User { Id = 12456L, UserName = "AnotherUser", IsActive = true };
+
+			CollectionAssert.DoesNotContain(
+				m_defaultUsers.Select( u => u.Id ),
+				user.Id
+			);
+
+			using( HttpResponseMessage response = await m_client.PostAsJsonAsync( "/api/users/", user ) ) {
+				Assert.AreEqual( HttpStatusCode.Created, response.StatusCode );
+			}
+
+			User actualUser = await m_usersContext.Users.FindAsync( user.Id );
+			Assert.NotNull( actualUser, "Could not find created user!" );
+			Assert.AreEqual( user.UserName, actualUser.UserName );
+			Assert.AreEqual( user.IsActive, actualUser.IsActive );
+		}
+
+		[Test]
+		public async Task CreateOrUpdate_UserExists_UpdatesUser() {
+			const string newUserName = "NewUserName";
+			User user = m_defaultUsers.First();
+			Assert.AreNotEqual( newUserName, user.UserName );
+
+			user.UserName = newUserName;
+			user.IsActive = !user.IsActive;
+
+			using( HttpResponseMessage response = await m_client.PostAsJsonAsync( "/api/users/", user ) ) {
+				Assert.AreEqual( HttpStatusCode.NoContent, response.StatusCode );
+			}
+
+			User actualUser = await m_usersContext.Users.FindAsync( user.Id );
+			Assert.NotNull( actualUser, "Could not find created user!" );
+			Assert.AreEqual( newUserName, actualUser.UserName );
+			Assert.AreEqual( user.IsActive, actualUser.IsActive );
+		}
+
+		[Test]
+		public async Task Delete_UserExists_DeletesUser() {
+			User userToDelete = m_defaultUsers.First();
+
+			using( HttpResponseMessage response = await m_client.DeleteAsync( $"/api/users/{userToDelete.Id}" ) ) {
+				Assert.AreEqual( HttpStatusCode.NoContent, response.StatusCode );
+			}
+
+			await m_usersContext.Entry( userToDelete ).ReloadAsync();
+			Assert.IsNull(
+				await m_usersContext.Users.FindAsync( userToDelete.Id ),
+				"User should have been deleted but was not!"
+			);
+		}
+
+		private Task AddUsers( params User[] users ) {
+			foreach( User user in users ) {
+				m_usersContext.Users.Add( user );

```

Comment:
For later: you can use `AddRange( users )` to avoid this loop.

Reply:
How much later? I want to make the change now :) .

Reply:
You make it whenever you want.  I just meant that it's not material to what this change is about, but you'll probably want to know/use it in the future :grin:.